### ISSUE TITLE
Could Build support?

### ISSUE THREAD

**Issue by slimshader** [modesttree/Projeny#16]
Hi,

sorry for asking here as it is not really an issue (didn't know where else to do it): does Projeny-based project support Unity Cloud Build service?

**Comment by svermeulen** [comment]
I can't say I've used Unity Cloud build but I strongly suspect that it isn't compatible.  This is because before doing your unity build, you would need to initialize the project using projeny, and so you'd have to run a custom script outside Unity, which probably isn't supported by the cloud build.

**Comment by sballew** [comment]
I've been trying to think of ways Cloud Build support could work. I'm thinking there could be something possible using the Pre-Export method.

> The fully-qualified name of a public static method you want us to call before we start the Unity build process. For example: ClassName.NeatMethod or NameSpace.ClassName.NeatMethod. No trailing parenthesis, and it can't have the same name as your Post-Export method!

Perhaps Projeny could generate a Unity project that includes only Projeny, and is checked into version control. Cloud Build could build this project and run some pre-export method that Projeny offers, which goes and sets up all the symlinks as necessary.

I have no idea if this is actually feasible within Unity Cloud Build. Some things that could go wrong are not being able to call any Projeny executables (Python?_, not being able to create symlinks, not knowing what platforms to build, and likely more.

Just brainstorming some thoughts here, as I do miss Cloud Build. The benefits of using Projeny outweigh the loss of Cloud Build, but a world with both would be great. I may take a stab at it to see if it is possible or not.

**Comment by svermeulen** [comment]
Yeah, I'm not sure how this would work either.  I wonder if you are even allowed to access the file system and create sym links etc. from C#.

**Comment by sballew** [comment]
Perhaps Projeny includes an option of doing a hard copy rather than symlinking. It wouldn't need to be used in general cases, but Cloud Build could use it to assemble the project at build time.

**Comment by svermeulen** [comment]
Some more info on a solution here:

https://forum.unity.com/threads/released-for-free-projeny-unity-project-and-package-manager.378544/page-2#post-3257380
